src/robusta/core/discovery/top_service_resolver.py

```diff
@@ -51,20 +51,29 @@ def store_cached_resources(cls, resources: List[TopLevelResource]):
     # TODO remove this guess function
     # temporary try to guess who the owner service is.
     @classmethod
-    def guess_service_key(cls, name: str, namespace: str) -> str:
-        resource = cls.guess_cached_resource(name, namespace)
+    def guess_service_key(cls, name: str, namespace: str, kind: str) -> str:
+        resource = cls.guess_cached_resource(name, namespace, kind=kind)
         return resource.get_resource_key() if resource else ""
 
     # TODO remove this guess function
     # temporary try to guess who the owner service is.
     @classmethod
-    def guess_cached_resource(cls, name: str, namespace: str) -> Optional[TopLevelResource]:
+    def guess_cached_resource(cls, name: str, namespace: str, kind: str,) \
+            -> Optional[TopLevelResource]:
         if name is None or namespace is None:
             return None
 
+        kind = kind.lower()
+
         for cached_resource in cls.__namespace_to_resource[namespace]:
-            if name.startswith(cached_resource.name):
-                return cached_resource
+            if kind == "pod" \
+                    or kind == "replicaset":
+                if name.startswith(cached_resource.name):
```

Comment:
maybe we can make this better 
like for pod named app-12346-sdfsgsdf
it will match with deployments named app-1, app-12 and app-123456
(with replica sets its less of an issue)
Pods have a replica set in the yaml, is there a way to use that?
<img width="430" alt="Screen Shot 2023-08-09 at 9 33 04" src="https://github.com/robusta-dev/robusta/assets/97387909/ce2d4f99-c9eb-46e9-813d-819f673e6ca3">

Reply:
maybe do something like this 
```
def get_pod_owner_reference(cls, name: str, namespace: str, kind: str)
if kind == "pod":
 robusta_pod = RobustaPod.find_pod(name, namespace)
 return robusta_pod.owner_references[0] if pobusta_pod.owner_references else none
```
and than in the function guess_cached_resource
```
...
 owner_reference = get_pod_owner_reference()
 if owner_reference:
    kind = owner_reference.kind.lower()
    name = owner_reference.name
for cached_resource in cls.__namespace_to_resource[namespace]:
...
```

Reply:
done